ad7768-1fmcz/ad77681.c

```diff
@@ -243,10 +313,321 @@ int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
 
 	/* Fill the adc_data buffer */
 	memcpy(adc_data, buf, ARRAY_SIZE(buf));
+	
+	if (buf[1] & 0x80)
+		converted_data = (int32_t)(0xFF << 24 | (((buf[1]  &   0xFF)  << 16) |	
```

Comment:
add brackets for `(0xFF << 24)` operation. Same as you did for `((buf[1]  &   0xFF)  << 16)`